### ISSUE TITLE
I don't think "tool-cache" cacheDir is doing anything

### ISSUE THREAD

**Issue by oprypin** [MeilCli/setup-crystal-action#3]
Hi. I found your usage of "[@actions/tool-cache](https://github.com/actions/toolkit/blob/master/packages/tool-cache/)" and it was very interesting to me, as it seems to promise caching of an action's downloads across different workflow runs. However, I noticed that it never gets any cache hits. For example, if you check [an example run](https://github.com/MeilCli/setup-crystal-action/runs/523915291) with "show timestamps", you will see that there is always a big delay before extracting the archive, so I think it's just downloading it every time. Did you ever see it being useful? After a long search I found that this is [usable only for self-hosted runners](https://github.com/actions/toolkit/blob/master/packages/tool-cache/README.md#cache), and also that [a proper caching feature for JavaScript is missing](https://github.com/actions/cache/issues/55).

**Comment by oprypin** [comment]
The code in question:

https://github.com/MeilCli/setup-crystal-action/blob/b4b36cb4e39d4303703b63f814368aabce8953be/src/setup_crystal.ts#L97 https://github.com/MeilCli/setup-crystal-action/blob/b4b36cb4e39d4303703b63f814368aabce8953be/src/setup_crystal.ts#L103

**Comment by MeilCli** [comment]
@oprypin 
Thanks question, I made it with reference to [setup-node](https://github.com/actions/setup-node) when making this Action.
I also noticed that it is downloaded every time, but I notice that can be expected to effective when multiple executing on Job or Runner.
Currently, that code is effective on self-hosted runners in most cases, but I think it is effective on ms-hosted-runner in some cases. So I put the code.
